### PR TITLE
stopping process when an article Id fails

### LINKED ISSUE #1329: method to stop scrapping when lot of article fail from rendering needed
while scrapping mdwiki [offliner command available here](https://farm.openzim.org/recipes/mdwiki/config)
due to slow mdwiki server rendering of many articles fails.

logs are provided in file given below

[2020-12-17.txt](https://github.com/openzim/mwoffliner/files/5705509/2020-12-17.txt)

### PR DESCRIPTION
**Pull request by MananJethwani**
@kelson42 this is to resolve #1329

### CODE REVIEW

**Comment by MananJethwani** [comment]
@kelson42 we need to change test and I think we should keep the concept of failed articles status as we will be defining a percentage or number of fails in fututre

File: src/Downloader.ts [hunk 1]
```diff
@@ -447,6 +447,9 @@ class Downloader {
     logger.info(`Getting article [${articleId}] from ${articleApiUrl}`);
 
     const json = await this.getJSON<any>(articleApiUrl);
+    if (json.error) {
```
Comment (kelson42) [inline src/Downloader.ts:450]: Why not putting this in getJSON()?

File: src/util/saveArticles.ts [hunk 2]
```diff
@@ -254,9 +254,8 @@ export async function saveArticles(zimCreator: ZimCreator, downloader: Downloade
                         dump.status.articles.success += 1;
                     }
                 } catch (err) {
-                    dump.status.articles.fail += 1;
-                    logger.warn(`Error downloading article [${articleId}], skipping`, err);
-                    await articleDetailXId.delete(articleId);
+                    logger.warn(`Error downloading article ${articleId}`);
```
Comment (kelson42) [inline src/util/saveArticles.ts:257]: This should be an error, not a warning

**Review by kelson42 — CHANGES_REQUESTED** [review]

File: src/Downloader.ts (hunk 1 above)
Comment (MananJethwani) [reply src/Downloader.ts:450]: getJSON is used for getting other things as well so if we put it there and we want to skip it that won't happen

File: src/util/saveArticles.ts (hunk 2 above)
Comment (MananJethwani) [reply src/util/saveArticles.ts:257]: ok

File: src/Downloader.ts (hunk 1 above)
Comment (kelson42) [reply src/Downloader.ts:450]: OK, but if I understand well, the `json` variable, does not have to be JSON? What about if this is `undefined`? Which part of the code set `json.error` (is that a javascript internal? Or just a value coming from the Mediawiki API?

File: src/util/saveArticles.ts [hunk 3]
```diff
@@ -254,9 +254,8 @@ export async function saveArticles(zimCreator: ZimCreator, downloader: Downloade
                         dump.status.articles.success += 1;
                     }
                 } catch (err) {
-                    dump.status.articles.fail += 1;
```
Comment (kelson42) [inline src/util/saveArticles.ts:257]: We should probably keep that line?!

**Review by kelson42 — CHANGES_REQUESTED** [review]
One of the thing which makes things unclear is the naming of methods I believe. What is the fundamental difference beetween `downloadContent()` and `getJSON()`? Why we don't have only one method?

**Review by kelson42 — APPROVED** [review]

**Pull request merged**